org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/symbols/LSPSymbolInFileDialog.java

```diff
@@ -78,33 +88,77 @@ protected Control createDialogArea(Composite parent) {
 		viewer.setLabelProvider(new NavigatorDecoratingLabelProvider(contentService.createCommonLabelProvider()));
 		viewer.setUseHashlookup(true);
 
-		final var isFirstSelectionEvent = new AtomicBoolean(true);
-		viewer.addSelectionChangedListener(event -> {
-			// ignore the first selection event which is fired when the current editor
-			// selection is pre-selected in the tree view when the outline popup is opens
-			if (isFirstSelectionEvent.get()) {
-				isFirstSelectionEvent.set(false);
-				return;
+		final Tree tree = viewer.getTree();
+
+		tree.addKeyListener(new KeyListener() {
+			@Override
+			public void keyPressed(KeyEvent e)  {
+				if (e.character == org.eclipse.swt.SWT.ESC) {
+					close();
+				}
 			}
+			@Override
+			public void keyReleased(KeyEvent e) {
+				// do nothing
+			}
+		});
+
+		// listen to enter key or other platform-specific default selection events
+		tree.addSelectionListener(new SelectionListener() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				// do nothing
+			}
+			@Override
+			public void widgetDefaultSelected(SelectionEvent e) {
+				gotoSelectedElement();
+			}
+		});
 
-			final var selection = (IStructuredSelection) event.getSelection();
-			if (selection.isEmpty()) {
-				return;
+		// listen to left mouse button's single click
+		tree.addMouseListener(new MouseAdapter() {
+			@Override
+			public void mouseUp(MouseEvent e) {
+				if (tree.getSelectionCount() < 1 || e.button != 1) {
+					return;
+				}
+
+				if (tree.equals(e.getSource())) {
+					TreeItem selection= tree.getSelection()[0];
+					if (selection.equals(tree.getItem(new Point(e.x, e.y)))) {
+						gotoSelectedElement();
+					}
+				}
 			}
+		});
+
+		this.getShell().addDisposeListener(new DisposeListener() {
 
-			Object item = selection.getFirstElement();
-			if (item instanceof final Either<?, ?> either) {
-				item = either.get();
+			@Override
+			public void widgetDisposed(DisposeEvent e) {
+				viewer = null;
 			}
+		});
+
+		viewer.setInput(outlineViewerInput);
+		return filteredTree;
+	}
+
+	private void gotoSelectedElement() {
+		Object selectedElement= getSelectedElement();
+
+		if (selectedElement != null) {
+			close();
```

Comment:
should it not go after the catch on the block contaning `textEditor.selectAndReveal(offset, endOffset - offset);`?

Reply:
Good question. My rationale was, if I want the editor to jump to a certain position and I have a user-confirmed selection in the quick outline view, i.e. the user did hit enter or clicked on the selected element (that are the only cases when `goToSelectedElement()` is called), then I want the quick outline view to be closed, even then if I detect some problem like a missing range. Otherwise, the view keeps being open and the user has no opportunity to close it (except of hitting ESC key).

Do you agree?

Reply:
Do you mean, I should keep the `close()` operation as last statement in the block `if (range != null) {...}` and create PR for moving it where it is now, after `if (selectedElement != null) {`?

Reply:
Yes, that would be great

Reply:
Ok, I change the close() call as you requested and created a PR #904 for moving the close operation as I suggested.